### ISSUE TITLE
Debugging fails on MacOS when plugins installed under $HOME/Library/Application Support

### ISSUE THREAD

**Issue by mgs255** [mbolotov/k6-intellij-plugin#16]
I was testing out the support for debugging K6 scripts the plugin fails on startup with the following error message reported in the process console:

```
/Users/michael.sommerville/.nvm/versions/node/v20.14.0/bin/node --experimental-default-type=module --experimental-specifier-resolution=node --disable-warning=ExperimentalWarning --import data:text/javascript,import { register } from 'node:module'; import { pathToFileURL } from 'node:url'; register('file:///Users/michael.sommerville/Library/Application Support/JetBrains/IntelliJIdea2023.3/plugins/k6/bundle/b.js', pathToFileURL('./').href); /Users/michael.sommerville/Library/Application Support/JetBrains/IntelliJIdea2023.3/plugins/k6/bundle/a.js
Debugger listening on ws://127.0.0.1:65248/6e8c778e-a799-4528-a3ac-4596d40b703e
For help, see: https://nodejs.org/en/docs/inspector
Debugger attached.
Waiting for the debugger to disconnect...
node:internal/modules/run_main:50
      throw err;
      ^

Error: ENOENT: no such file or directory, lstat '/Users/michael.sommerville/Library/Application'
    at Object.realpathSync (node:fs:2707:29)
    at toRealPath (node:internal/modules/helpers:58:13)
    at resolveMainPath (node:internal/modules/run_main:43:18)
    at Object.executeUserEntryPoint (node:internal/modules/run_main:160:24)
    at node:internal/main/run_main_module:19:40 {
  errno: -2,
  code: 'ENOENT',
  syscall: 'lstat',
  path: '/Users/michael.sommerville/Library/Application'
}

Node.js v20.14.0

Process finished with exit code 1
```

**Comment by mbolotov** [comment]
Hi Michael,
Thanks for reporting this problem!
Could you try this bundle with the fix?
[k6-1.2.1-EAP1.zip](https://github.com/user-attachments/files/16904053/k6-1.2.1-EAP1.zip)

**Comment by mgs255** [comment]
That works much better, thank you!  

Some additional feedback for you.  My script was using import https notation from the [example](https://grafana.com/docs/k6/latest/javascript-api/jslib/utils/), e.g:

```
import { sleep } from 'k6';
import http from 'k6/http';

import {
  randomIntBetween,
  randomString,
  randomItem,
  uuidv4,
  findBetween,
} from 'https://jslib.k6.io/k6-utils/1.4.0/index.js';

export default function () {
  const res = http.post(`https://test-api.k6.io/user/register/`, {
    first_name: randomItem(['Joe', 'Jane']), // random name
    last_name: `Jon${randomString(1, 'aeiou')}s`, //random character from given list
    username: `user_${randomString(10)}@example.com`, // random email address,
    password: uuidv4(), // random password in form of uuid
  });

  // find a string between two strings to grab the username:
  const username = findBetween(res.body, '"username":"', '"');
  console.log('username from response: ' + username);

  sleep(randomIntBetween(1, 5)); // sleep between 1 and 5 seconds.
}
```

This was failing with an error related to the import prefix not being supported.  In order to make that work in IntelliJ I had to download k6-utils locally and import it like so:
`import { randomItem, randomString, uuidv4 } from './src/utils.js';`

Furthermore, even though the debug now works correctly, 'run' button seems to be broken, giving the following error in a popup window only:

```
Failed to start [k6, run, /Users/michael.sommerville/Work/generic-sso-load-test/test.js] in /, [null], envs={intellij_plugin_wrapper_host=192.168.1.61, PATH=/usr/bin:/bin:/usr/sbin:/sbin, __CFBundleIdentifier=com.jetbrains.intellij, SHELL=/bin/zsh, IJ_RESTARTER_LOG=/Users/michael.sommerville/Library/Logs/JetBrains/IntelliJIdea2023.3/restarter.log, USER=michael.sommerville, TMPDIR=/var/folders/5l/b98yxyqx55q8753cll70f7pw0000gp/T/, COMMAND_MODE=unix2003, SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.wxUo7BFi6O/Listeners, XPC_FLAGS=0x0, __CF_USER_TEXT_ENCODING=0x1F6:0x0:0x0, LOGNAME=michael.sommerville, XPC_SERVICE_NAME=application.com.jetbrains.intellij.1681011.2018598, IDEA_INITIAL_DIRECTORY=/, TOOLBOX_VERSION=2.4.2.32922, HOME=/Users/michael.sommerville}
```

The restarter log doesn't seem helpful:

```
============= 2024-09-06 07:34:05.232575 UTC
    2 [info ] args: ["/Users/michael.sommerville/Applications/IntelliJ IDEA Ultimate.app/Contents/bin/restarter", "98450", "2", "/usr/bin/open", "/Users/michael.sommerville/Applications/IntelliJ IDEA Ultimate.app"]
    3 [info ] CWD: Ok("/Users/michael.sommerville")
    3 [info ] waiting for 98450
  615 [info ] exec-ing: ["/usr/bin/open", "/Users/michael.sommerville/Applications/IntelliJ IDEA Ultimate.app"]
```

Finally, there was another issue I ran into where it looks like the plugin requires a `default` function.  Our test was using
the scenarios functionality and there was no default function which caused the plugin to error.

**Comment by mbolotov** [comment]
Thanks for all the feedback.
I miss them so much. The plugin has been downloaded thousands of times, but there are almost no feedbacks.

There seems no problem to add to add an ability to download sources from remote location for Node.js (debug mode). 
However, it the k6 `utils.js` has been rewritten for Common JS model since version `1.3` and it currently cannot be executed by Node.js (in the way as it launched by the plugin). I will try to resolve this problem.

I was unable to reproduce the problem with locally downloaded `utils.js`. k6 runs it successfully. Could you provide the entire project?

As for the `default` function issue - could you provide the setup? Do you define the scenario name in a command line parameter? Current version of the plugin searches for the scenario name within the configuration defined in the script code only .

**Comment by mbolotov** [comment]
I've added a hook to process remote imports: [k6-1.2.1-EAP2.zip](https://github.com/user-attachments/files/16911177/k6-1.2.1-EAP2.zip)

Your example works okay with debugger if the version `k6-utils` is downgraded to `1.2`
```
import { sleep } from 'k6';
import http from 'k6/http';

import {
  randomIntBetween,
  randomString,
  randomItem,
  uuidv4,
  findBetween,
} from 'https://jslib.k6.io/k6-utils/1.2.0/index.js';

export default function () {
  const res = http.post(`https://test-api.k6.io/user/register/`, {
    first_name: randomItem(['Joe', 'Jane']), // random name
    last_name: `Jon${randomString(1, 'aeiou')}s`, //random character from given list
    username: `user_${randomString(10)}@example.com`, // random email address,
    password: uuidv4(), // random password in form of uuid
  });

  // find a string between two strings to grab the username:
  const username = findBetween(res.body, '"username":"', '"');
  console.log('username from response: ' + username);

  sleep(randomIntBetween(1, 5)); // sleep between 1 and 5 seconds.
}
```

**Comment by mbolotov** [comment]
Hi @mgs255!
Have you had a chance to try out the k6-1.2.1 version?

**Comment by mbolotov** [comment]
> Have you had a chance to try out the k6-1.2.1 version?

@mgs255 ?

**Comment by mbolotov** [comment]
original issue has been fixed in `1.2.1` version
